src/main/default/classes/TriggerHandler.cls

```diff
@@ -65,51 +73,66 @@ public class TriggerHandler {
 
         if (handlers != null && !handlers.isEmpty()) {
             for (HandlerInterface h : handlers) {
-                if (!handlerDisabled(h) && !objectHandlerDisabled(sObjectType, h)) {
+                if (!handlerDisabled(h)) {
                     h.handle();
                 }
             }
         }
     }
 
-    private static Set<String> disabledObjects = new Set<String>();
-    private static Set<String> disabledHandlers = new Set<String>();
-    private static Map<String, Set<String>> disabledObjectHandlers = new Map<String, Set<String>>();
-    @TestVisible
-    private static void cacheTriggerControl(Trigger_Control__mdt[] triggerControls) {
-        disabledObjects.clear();
-        disabledHandlers.clear();
-        disabledObjectHandlers.clear();
-        for (Trigger_Control__mdt triggerControl : triggerControls) {
-            String objectName = triggerControl.Object__c == null ? null : triggerControl.Object__c.toLowerCase();
-            String handlerName = triggerControl.Class__c == null ? null : triggerControl.Class__c.toLowerCase();
-            if (handlerName == null && objectName != null) {
-                disabledObjects.add(objectName);
-            } else if (handlerName != null && objectName == null) {
-                disabledHandlers.add(handlerName);
-            } else if (handlerName != null && objectName != null) {
-                if (!disabledObjectHandlers.containsKey(objectName)) {
-                    disabledObjectHandlers.put(objectName, new Set<String>());
-                }
-                disabledObjectHandlers.get(objectName).add(handlerName);
+    private void doAutomaticBinding() {
+        if (!triggerConfig.enabled) {
+            return;
+        }
+        for (String handlerName : triggerConfig.handlerOrder) {
+            TriggerConfigService.HandlerConfig handlerConfig = triggerConfig.handlers.get(handlerName);
+            // do not bind if auto bind is off or if handler is disabled
+            if (!handlerConfig.autoBind || !handlerConfig.enabled) {
```

Comment:
do we need an autoBind variable? if they don't want to auto-bind they can just disable the handler, seems redundant and less is better with frameworks ...

Reply:
I believe this was to allow for disabling of statically bound handlers, via the same configuration

Reply:
EG:

Don't create and bind, because I've it's already setup statically in the trigger...

It is possible that the "Auto-Bind" routine could just check to see if it's already statically added, but that only works if a trigger handler instance can only be bound once (this would only happen if you have the same class being reused with different construction parameters for some reason)

Reply:
yep, that's why it was there, so my fault this is in here, but i think we can live without this ...

Reply:
So we do implement the "first check for static bind" or just not allow statically bound triggers to be disabled?

Reply:
uggh, so i missed during the first pass that we'd gotten rid of the old trigger control that allowed static bound handlers to be disabled. this whole conversation and the need for auto bind is making a lot more sense now. i'm torn, but gun to my head i think starting with just not allowing statically bound triggers to be disabled makes sense, it adds a bit of stick to encourage switching to dynamic bindings, and there is still the option to disable things at the object level.

that said, there are valid use cases where constructor parameters are very helpful, so maybe i've got this wrong. struggling a bit on what the least surprise behavior is here ...

big picture figure it's super easy to bloat this sort of thing, so better to start with as small a data model and use case set as we can conceivably live with

Reply:
In theory the need for inputs can easily be factored away with composition.

IOW, the class you dynamically bind is just a wrapper for the "reusable" class that setups the inputs.

I think the most "elegant" solution (from a configuration POV) would be to just check for statically bound handler and not inject if it already exists.

However, this would come at the cost of a functional constraint, a small amount of resources and some additional code.

The `Auto-Bind` achieves the same thing goal without the above penalties, but does so in a way that is kinda weird and unintuitive from a configuration POV.

The third option of just "not supporting disablement of statically bound handlers" is somewhat disappointing as that means the before you really get the benefits from this framework, you have to refactor.

Reply:
I think I like the auto bind from a user standpoint I think.

If we don't have auto bind (which is defaulting to true and has some help text), we then have to delete any MDT records if we want to toggle something back on.

And like Charlie is saying - this way we don't have to check for what has been bound.

Reply:
i'm sold (at least i hope), so you're thinking
- ditch the auto-bind variable
- only bind an instance of a handler class once for an object/operation

Reply:
@jake-hebert a little extra code to shrink the data model seems worth it to me ...

Reply:
My brain is catching up. Minimal changes required to ditch the auto-bind field.

Reply:
@jake-hebert you would need to look at what is statically bound and NOT rebind anything dynamically that is already there.

Reply:
Because we're allowing direct binding, trying to handle this is the auto bind doesn't make sense because that would always happen first when the class gets constructed. So auto bind would run and then there could be a manual binding as well.

Could check every time a bind happens I guess.

For now, just handling this by keeping track of what has been run on each event and only allowing one run. Pretty much same/same I think.